Sources/UI/Channels/ChannelTableViewCell.swift

```diff
@@ -86,48 +94,68 @@ public final class ChannelTableViewCell: UITableViewCell, Reusable {
         infoLabel.textColor = style.dateColor
         infoLabel.backgroundColor = backgroundColor
         
-        contentView.addSubview(avatarView)
         contentView.addSubview(nameLabel)
         contentView.addSubview(messageLabel)
         contentView.addSubview(dateLabel)
         contentView.addSubview(infoLabel)
-
-        avatarView.backgroundColor = backgroundColor
-        avatarView.snp.makeConstraints { make in
-            make.top.equalToSuperview().offset(CGFloat.messageInnerPadding).priority(999)
-            make.left.equalToSuperview().offset(CGFloat.messageEdgePadding)
-            make.size.equalTo(CGFloat.channelBigAvatarSize)
+        var hasAvatar = false
+        
+        if let avatarViewStyle = style.avatarViewStyle {
+            hasAvatar = true
+            contentView.addSubview(avatarView)
+            
+            avatarView.backgroundColor = backgroundColor
+            avatarView.snp.makeConstraints { make in
+                make.top.equalToSuperview().offset(style.edgeInsets.top).priority(999)
+                make.left.equalToSuperview().offset(style.edgeInsets.left)
+                make.size.equalTo(avatarViewStyle.size)
+            }
         }
         
         nameLabel.snp.makeConstraints { make in
-            make.left.equalTo(avatarView.snp.right).offset(CGFloat.messageInnerPadding)
-            make.right.lessThanOrEqualTo(dateLabel.snp.left).offset(-CGFloat.messageInnerPadding)
-            make.bottom.equalTo(avatarView.snp.centerY)
+            if hasAvatar {
+                make.bottom.equalTo(avatarView.snp.centerY).offset(style.spacing.vertical / -2).priority(999)
+                make.left.equalTo(avatarView.snp.right).offset(style.spacing.horizontal)
```

Comment:
Not for this PR, but we should start using `leading` and `trailing` for labels, so in the future we can support right to left languages easily.

Reply:
A good point.